src/porepy/numerics/nonlinear/line_search.py

```diff
@@ -18,7 +18,7 @@ class ConstraintLineSearchNonlinearSolver(
         LineSearchNewtonSolver,
     ):
         pass
-    model.params["nonlinear_solver"] = ConstraintLineSearchNonlinearSolver
+    solver_params["nonlinear_solver"] = ConstraintLineSearchNonlinearSolver
```

Comment:
Please add an illustration of how the solver_params should be passed. Please also fix "model parameters"  to "solver parameters" in lines 12 and 24.